app/models/listing_shape.rb

```diff
@@ -35,6 +35,8 @@ class ListingShape < ActiveRecord::Base
     :price_quantity_placeholder,
   )
 
+  validates_uniqueness_of :name, :scope => :community_id
```

Comment:
Good suggestion! This is not validated with unique index. I guess the model selects all shapes by `community_id`.